frame/support/procedural/src/construct_runtime/expand/inherent.rs

```diff
@@ -90,9 +90,37 @@ pub fn expand_outer_inherent(
 				use #scrate::inherent::{ProvideInherent, IsFatalError};
 				use #scrate::traits::{IsSubType, ExtrinsicCall};
 				use #scrate::sp_runtime::traits::Block as _;
+				use #scrate::_private::sp_inherents::Error;
+				use #scrate::log;
 
 				let mut result = #scrate::inherent::CheckInherentsResult::new();
 
+				// This handle assume we abort on the first fatal error.
+				fn handle_put_error_result(res: Result<(), Error>) {
+					const LOG_TARGET: &str = "runtime::inherent";
+					match res {
+						Ok(()) => (),
+						Err(Error::InherentDataExists(id)) =>
+							log::debug!(
+								target: LOG_TARGET,
+								"Some error already reported for inherent {:?}, new non fatal \
+								error is ignored",
+								id
+							),
+						Err(Error::FatalErrorReported) =>
+							log::error!(
+								target: LOG_TARGET,
+								"Fatal error already reported, unexpected considering there is \
+								only one fatal error",
+							),
+						Err(_) =>
```

Comment:
error variant are different when sp-inherents is compiled with `std` or not.
but some test crate use `construct_runtime` without a `std` feature. So there is no way to know what is the actual error type.